### ISSUE TITLE
Unable to use ordinal in ModifyVariable while index works

### ISSUE THREAD

**Issue by The-PPWD** [SpongePowered/Mixin#481]
I experienced a potential bug with a ModifyVariable mixin that goes as follows:

The method I'm mixing into:
```java

      public void updateCameraPosition(ServerPlayerEntity player) {
         if (player != this.entity) {
            Vec3d vec3d = player.getPos().subtract(this.entry.getLastPos());
            int i = Math.min(this.getMaxTrackDistance(), (ThreadedAnvilChunkStorage.this.watchDistance - 1) * 16);
            boolean bl = vec3d.x >= (double)(-i) && vec3d.x <= (double)i && vec3d.z >= (double)(-i) && vec3d.z <= (double)i && this.entity.canBeSpectated(player);
            if (bl) {
               boolean bl2 = this.entity.teleporting;
               if (!bl2) {
                  ChunkPos chunkPos = new ChunkPos(this.entity.chunkX, this.entity.chunkZ);
                  ChunkHolder chunkHolder = ThreadedAnvilChunkStorage.this.getChunkHolder(chunkPos.toLong());
                  if (chunkHolder != null && chunkHolder.getWorldChunk() != null) {
                     bl2 = ThreadedAnvilChunkStorage.getChebyshevDistance(chunkPos, player, false) <= ThreadedAnvilChunkStorage.this.watchDistance;
                     
                     // Where I want the ModifyVariable at
                     
                  }
               }

               if (bl2 && this.playersTracking.add(player)) {
                  this.entry.startTracking(player);
               }
            } else if (this.playersTracking.remove(player)) {
               this.entry.stopTracking(player);
            }

         }
      }
```

Since I want the mixin to apply after the second `STORE` instruction on the second `boolean` variable, my annotation looked like this:

```java
    @ModifyVariable(
            method = "updateCameraPosition(Lnet/minecraft/server/network/ServerPlayerEntity;)V",
            at = @At(
                    value = "STORE",
                    ordinal = 1
            ),
            ordinal = 1
    )
```

However, this caused a seemingly infinite loop of mixin apply failures:

```java
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: Injection validation failed: Variable modifier method ... expected 1 invocation(s) but 0 succeeded.
```

However, changing `ordinal = 1` to `index = 5` fixed the issue.

```java
    @ModifyVariable(
            method = "updateCameraPosition(Lnet/minecraft/server/network/ServerPlayerEntity;)V",
            at = @At(
                    value = "STORE",
                    ordinal = 1
            ),
            index = 5
    )
```

Something that's important to note is that I can use `ordinal = 1` and `index = 5` if I target the first `STORE` instruction of the second `boolean` without issue; however, only `index = 5` works when targeting the second `STORE` instruction.

If there's more information that is needed, please let me know.

**Comment by Mumfrey** [comment]
I can't replicate the described behaviour with the latest snapshot (currently 0.8.3) and MCP, with your first example which supposedly fails transcribed to MCP mappings and with `print` enabled to emit the matches:
```java
@Mixin(targets = "net/minecraft/world/server/ChunkManager$EntityTracker")
class MixinEntityTracker {
    @ModifyVariable(
        method = "updateTrackingState(Lnet/minecraft/entity/player/ServerPlayerEntity;)V",
        at = @At(
            value = "STORE",
            ordinal = 1
        ),
        ordinal = 1,
        print = true
    )
    private boolean modifyBoolean(boolean orginalValue) {
        return orginalValue;
    }
}
```
I see the expected result:
```python
/*****************************************************************************************************************/
/*         Target Class : net.minecraft.world.server.ChunkManager$EntityTracker                                  */
/*        Target Method : updateTrackingState                                                                    */
/*        Callback Name : modifyBoolean                                                                          */
/*         Capture Type : boolean                                                                                */
/*          Instruction : [128] LabelNode UNKNOWN                                                                */
/*****************************************************************************************************************/
/*           Match mode : EXPLICIT (match by criteria)                                                           */
/*        Match ordinal : 1                                                                                      */
/*          Match index : any                                                                                    */
/*        Match name(s) : any                                                                                    */
/*            Args only : false                                                                                  */
/*****************************************************************************************************************/
/* INDEX  ORDINAL                            TYPE  NAME                                                CANDIDATE */
/* [  1]    [  0]              ServerPlayerEntity  p_219400_1_                                         -         */
/* [  2]    [  0]                        Vector3d  vector3d                                            -         */
/* [  3]    [  0]                             int  i                                                   -         */
/* [  4]    [  0]                         boolean  flag                                                YES       */
/* [  5]    [  1]                         boolean  flag1                                               YES       */
/* [  6]                                        -                                                                */
/* [  7]                                        -                                                                */
/*****************************************************************************************************************/
```
Changing the `ordinal = 1` to `index = 5` produces the same (expected) result. Please show the results of `print = true` with your injection which fails.

**Comment by The-PPWD** [comment]
Here is my print when using `index = 5`. Note that I'm unable to get it to print when using `ordinal = 1` as it throws the same error as before. Mixin version is 0.8.2.

```
/*****************************************************************************************************************/
/*         Target Class : net.minecraft.server.world.ThreadedAnvilChunkStorage$EntityTracker                     */
/*        Target Method : updateCameraPosition                                                                   */
/*        Callback Name : updateCameraPosition_STORE1                                                            */
/*         Capture Type : boolean                                                                                */
/*          Instruction : [138] LabelNode UNKNOWN                                                                */
/*****************************************************************************************************************/
/*           Match mode : EXPLICIT (match by criteria)                                                           */
/*        Match ordinal : any                                                                                    */
/*          Match index : 5                                                                                      */
/*        Match name(s) : any                                                                                    */
/*            Args only : false                                                                                  */
/*****************************************************************************************************************/
/* INDEX  ORDINAL                            TYPE  NAME                                                CANDIDATE */
/* [  1]    [  0]              ServerPlayerEntity  player                                              -         */
/* [  2]                                        -                                                                */
/* [  3]                                        -                                                                */
/* [  4]                                        -                                                                */
/* [  5]    [  0]                         boolean  bl2                                                 YES       */
/* [  6]                                        -                                                                */
/* [  7]                                        -                                                                */
/*****************************************************************************************************************/
```

**Comment by Mumfrey** [comment]
So investigating this a bit deeper, this has the same root cause as #482 and should have actually been fixed by 6110084d0bc9404ebd085e91e23c17266dc2a5c7. However testing those fixes in a fabric environment actually uncovered some flaws in the fix, so I have implemented a fix for the fix which should hopefully nail this once and for all. Though I've said that several times over the years so I'm not going to hold my breath.

However this has actually had a neat side-effect of revealing more variables which the analyser can be "confident" are in-scope, so in the case of this example actually reveals the `chunkPos` and `chunkHolder` variables as well (since they are in scope at the capture point):

```python
/*****************************************************************************************************************/
/*         Target Class : net.minecraft.server.world.ThreadedAnvilChunkStorage$EntityTracker                     */
/*        Target Method : updateCameraPosition                                                                   */
/*        Callback Name : updateCameraPosition_STORE1                                                            */
/*         Capture Type : boolean                                                                                */
/*          Instruction : [132] LabelNode UNKNOWN                                                                */
/*****************************************************************************************************************/
/*           Match mode : EXPLICIT (match by criteria)                                                           */
/*        Match ordinal : 1                                                                                      */
/*          Match index : any                                                                                    */
/*        Match name(s) : any                                                                                    */
/*            Args only : false                                                                                  */
/*****************************************************************************************************************/
/* INDEX  ORDINAL                            TYPE  NAME                                                CANDIDATE */
/* [  1]    [  0]              ServerPlayerEntity  player                                              -         */
/* [  2]    [  0]                           Vec3d  vec3d                                               -         */
/* [  3]    [  0]                             int  i                                                   -         */
/* [  4]    [  0]                         boolean  bl                                                  YES       */
/* [  5]    [  1]                         boolean  bl2                                                 YES       */
/* [  6]    [  0]                        ChunkPos  chunkPos                                            -         */
/* [  7]    [  0]                     ChunkHolder  chunkHolder                                         -         */
/*****************************************************************************************************************/
```
So hopefully this will provide some meaningful improvements to localcapture on all platforms, fabric included.
